routes/api/users.js

```diff
@@ -0,0 +1,264 @@
+const express = require('express');
+const router = express.Router();
+const bcrypt = require('bcryptjs');
+var userModel = require('../../models/user');
+const itineraryModel = require('../../models/itinerary');
+const jwt = require("jsonwebtoken");
+const keys = require("../../config/keys");
+const passport = require('passport');
+
+/*get all users*/
+router.get("/all",
+    /* Uncomment next line to add web token athentification */
+    //passport.authenticate("jwt", { session: false }),
+    (req, res) => {
+        userModel.find({})
+            .then(users => {
+                res.json(
+                    users.map(e => {
+                        return {
+                            id: e.id,
+                            name: e.name,
+                            email: e.email
+                        }
+                    }
+                    ))
+            })
+            .catch(err => res.status(404).json({ error: "No users" }));
+    }
+);
+
+/*get user by ID*/
+router.get('/:id',
+    /* Uncomment next line to add web token athentification */
+    //passport.authenticate("jwt", { session: false }),
```

Comment:
This could probably get refactored to be more reusable as a sugar method and it being enabled or disabled through an environment variable.